src/figuresClasses.ts

```diff
@@ -1,19 +1,76 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape:Shape;
+  color:Color;
+  getArea():number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape:Shape ='triangle';
 
-}
+  constructor(
+    public color:Color,
+    public a:number,
+    public b:number,
+    public c:number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('The side is a positive value');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error(`The longest side of a triangle cannot be greater
+        than the sum of other two`);
+    }
+  }
 
-export class Circle {
+  getArea():number {
+    const p :number = (this.a + this.b + this.c) / 2;
+    const s :number = p * (p - this.a) * (p - this.b) * (p - this.c);
 
+    return Math.floor(Math.sqrt(s) * 100) / 100;
```

Comment:
```suggestion
    const { a, b, c } = this;
    const halfPerimeter = (a + b + c) / 2;
    const area = p * (p - a) * (p - b) * (p - c);

    return Math.floor(Math.sqrt(s) * 100) / 100;
```
consider to destructure this.
better to make more meaningful naming.
redundant to set directly type for halfPerimeter and area.
typescript knows that a, b, c are numbers and that result will be a number.